app/src/test/java/org/mozilla/fenix/widget/VoiceSearchActivityTest.kt

```diff
@@ -141,50 +134,6 @@ class VoiceSearchActivityTest {
         assertNull(shadow.peekNextStartedActivityForResult())
     }
 
-    @Test
-    fun `handle speech result`() {
```

Comment:
Couldn't we adapt the tests to the new API? Just removing them seems a bit risky.

Reply:
I spent most of my time on trying to get these two tests working.  However, with the new implementation, `shadow.receiveResult()` doesn't work anymore.  I couldn't find another way to perform the tests here and had to remove them.

Reply:
Yeah, I couldn't find any way to make `ShadowActivity` work with the new API as well. 

Maybe we could do without it? It looks like the tests are checking the code that handles the [activity result](https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/widget/VoiceSearchActivity.kt#89) – if the new activity is called and the current is closed. 

We can't make the system pass the call through the new API, but we still could test the code that should work after the system call. We could move the code that used to be in `onActivityResult` into a separate, internal and visible for testing function, and call it from on `registerForActivityResult` and from the tests as well!

That way we could keep the tests, just changing one line – `shadow.receiveResult` to something like `activity.handleActivityResult` :)

Reply:
That's a good idea.  I'll update.  Thanks